kernel/kernel-api/src/main/java/io/delta/kernel/internal/util/ColumnMapping.java

```diff
@@ -249,38 +283,126 @@ private static boolean allowMappingModeChange(String oldMode, String newMode) {
      */
     private static Metadata assignColumnIdAndPhysicalName(Metadata metadata, boolean isNewTable) {
         StructType schema = metadata.getSchema();
-        int maxColumnId = Math.max(
+        AtomicInteger maxColumnId = new AtomicInteger(
+                Math.max(
                 Integer.parseInt(metadata.getConfiguration()
                         .getOrDefault(COLUMN_MAPPING_MAX_COLUMN_ID_KEY, "0")),
```

Comment:
indentation

Reply:
it would be great to eventually have an automatic code formatter in the codebase to avoid having to manually re-structure code

Reply:
yes I've introduced code formatters to projects like Iceberg and UnityCatalog (and a bunch of others) and can do that as well here